src/arvgvfakecamera.c

```diff
@@ -493,14 +501,27 @@ arv_gv_fake_camera_start (ArvGvFakeCamera *gv_fake_camera)
 		    (ifap_iter->ifa_flags & IFF_POINTOPOINT) == 0 &&
 		    (ifap_iter->ifa_addr->sa_family == AF_INET) &&
 		    g_strcmp0 (ifap_iter->ifa_name, gv_fake_camera->priv->interface_name) == 0) {
+#else
+	ifaces=arv_enumerate_network_interfaces();
+	if(!ifaces){
+		arv_warning_device ("[GvFakeCamera::start] No network interface found");
+		return FALSE;
+	}
+
+	for (iface_iter=ifaces; iface_iter!=NULL; iface_iter=iface_iter->next){ {
+#endif
 			GSocketAddress *socket_address;
 			GInetAddress *inet_address;
 			GInetAddress *gvcp_inet_address;
 			unsigned int n_socket_fds;
 			unsigned int i;
-
+#ifndef _ARV_IFACES
 			socket_address = g_socket_address_new_from_native (ifap_iter->ifa_addr,
 									   sizeof (struct sockaddr));
+#else
+			socket_address = g_socket_address_new_from_native ((struct sockaddr*)((ArvIfaceAddr*)iface_iter->data)->addr,
```

Comment:
Looks like struct sockaddr * cast is redundant. I would also prefer a getter function for addr.

Reply:
For the getter function, you have something like `arv_network_interface_get_{addr,netmask, broadaddr,name}` for the data members in mind?

Reply:
Yes.